src/theory/arith/theory_arith_private.cpp

```diff
@@ -4456,12 +4459,21 @@ bool TheoryArithPrivate::rowImplicationCanBeApplied(RowIndex ridx, bool rowUp, C
         std::vector<Node> farkasCoefficients;
         farkasCoefficients.reserve(coeffs->size());
         auto nm = NodeManager::currentNM();
-        std::transform(coeffs->begin(),
-                       coeffs->end(),
-                       std::back_inserter(farkasCoefficients),
-                       [nm](const Rational& r) {
-                         return nm->mkConst<Rational>(CONST_RATIONAL, r);
-                       });
+        if (type.isInteger())
+        {
+          std::transform(coeffs->begin(),
+                         coeffs->end(),
+                         std::back_inserter(farkasCoefficients),
+                         [nm](const Rational& r) { return nm->mkConstInt(r); });
+        }
+        else
+        {
+          std::transform(
+              coeffs->begin(),
+              coeffs->end(),
+              std::back_inserter(farkasCoefficients),
+              [nm](const Rational& r) { return nm->mkConstReal(r); });
+        }
```

Comment:
```suggestion
        std::transform(coeffs->begin(),
                       coeffs->end(),
                       std::back_inserter(farkasCoefficients),
                       [nm, type](const Rational& r) {
                         return nm->mkConstRealOrInt(type, r);
                       });
```

Reply:
This doesn't work, I'm not sure why. It complains that `type` is not bound.